src/modeler/AutoRest.Swagger/Validation/Extensions/XmsClientNameProperty.cs

```diff
@@ -41,6 +41,18 @@ public class XmsClientNameProperty : TypedRule<Dictionary<string, Schema>>
         public override Category Severity => Category.Error;
 
         /// <summary>
+        /// What kind of open api document type this rule should be applied to
+        /// </summary>
+        public override ServiceDefinitionDocumentType ServiceDefinitionDocumentType => ServiceDefinitionDocumentType.ARM;
```

Comment:
any generated sdk can leverage this. On that bases, I'd be inclined for `default`